src/App.tsx

```diff
@@ -1,27 +1,118 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+import User from './types/User';
+import Todo from './types/Todo';
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [currentTodos, setCurrentTodos] = useState<Todo[]>(todos);
+
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorSelectedUserId, setErrorSelectedUserId] = useState(false);
+
+  function getId(todosArr: Todo[]): number {
+    const sortedTodos = [...todosArr].sort((a, b) => b.id - a.id);
+
+    return sortedTodos[0].id + 1;
+  }
+
+  const addTodo = (newTodo: Todo) => {
+    setCurrentTodos(prevTodos => [...prevTodos, newTodo]);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (!title) {
+      setErrorTitle(true);
+    }
+
+    if (!selectedUserId) {
+      setErrorSelectedUserId(true);
+    }
+
+    if (!title || !selectedUserId) {
+      return;
+    }
+
+    addTodo({
+      id: getId(currentTodos),
+      title: title,
+      userId: selectedUserId,
+      completed: false,
+      user: getUserById(selectedUserId),
+    });
+
+    setTitle('');
+    setSelectedUserId(0);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label className="label" htmlFor="titleInput">
+            Title:&nbsp;
+            <input
+              type="text"
+              data-cy="titleInput"
+              id="titleInput"
+              placeholder="Enter a title"
+              value={title}
+              onChange={event => {
```

Comment:
If your onChange element uses more than one function the best practice is to create a handler.  Something like that:
const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setTitle(event.target.value);
    setErrorTitle(false);
  };